src/components/home/ImportChallengeButton.tsx

```diff
@@ -0,0 +1,63 @@
+import { HomeCard } from "./HomeCard";
+import ImportImage from "../../assets/import.png"
+import Button from '@mui/material/Button';
+import { Ember } from "../../emberCommunication";
+import { useNavigate } from "react-router-dom";
+import simpleTypeGuard, { SimpleArray, SimpleNumber, SimpleString } from 'simple-type-guard';
+import { useState } from "react";
+import { Modal } from "@mui/material";
+
+export type ImportedChallenge = {
+    version: number,
+    nombre: string,
+    escena: string,
+    bloques: string[]
+}
+
+export const ImportChallengeCard = () => {
+    const navigate = useNavigate();
+    const [modalOpen, setModalOpen] = useState(false);
+
+    const goToChallenge = (challenge: ImportedChallenge) => {
+        Ember.importChallenge(challenge)
+        navigate("/desafioImportado")
+    }
+
+    const showErrorModal = () => {
+        setModalOpen(true)
+    }
+
+    const isValidChallenge = (json: unknown): boolean => 
+        simpleTypeGuard<ImportedChallenge>(json, {
+            version: SimpleNumber, 
+            nombre: SimpleString, 
+            escena: SimpleString, 
+            bloques: new SimpleArray(SimpleString),
+        })
+
+    const readFile = async (event: any) => {
+        const file: File = event.target.files[0]
+        const content: string = await file.text()
+        const challengeJson: unknown = JSON.parse(content)
+
+        event.target.value = null // Without this Chrome seems to cache the file and prevents reruns of this function. 
+
+        if (isValidChallenge(challengeJson)) {
+            goToChallenge(challengeJson as ImportedChallenge)
+        }
+        else {
+            showErrorModal()
+        }
+
+    }
+
+    return (
+    <Button component="label" style={{textTransform: 'none'}}>
+        <HomeCard nameKey={"import"} image={ImportImage} color={"#fc3e5e"}/>
+        <input hidden accept=".json" type="file" onChange={readFile}/>
```

Comment:
Propongo una extensión (que no existe en internet....)

```suggestion
        <input hidden accept=".pbch,.json" type="file" onChange={readFile}/>
```

Me pregunto qué tan difícil será a futuro aceptar zips y distinguir si lo que me vino es json o zip. Tengo ganas de el año que viene continuar lo que empezó tomi freilij.